rust-runtime/aws-smithy-observability/src/global.rs

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Utilities for interacting with the currently set `GlobalTelemetryProvider`
+
+use once_cell::sync::Lazy;
+use std::{
+    mem,
+    sync::{Arc, RwLock},
+};
+
+use crate::provider::{GlobalTelemetryProvider, TelemetryProvider};
+
+// Statically store the global provider
+static GLOBAL_TELEMETRY_PROVIDER: Lazy<RwLock<GlobalTelemetryProvider>> =
+    Lazy::new(|| RwLock::new(GlobalTelemetryProvider::new(TelemetryProvider::default())));
+
+/// Set the current global [TelemetryProvider]. If [None] is supplied then a noop provider is set.
+/// The previous [TelemetryProvider] is returned in an [Arc] so appropriate cleanup can be done if necessary.
+pub fn set_global_telemetry_provider(
```

Comment:
Given the namespacing of this in module `global` we can probably drop `_global_` from the name (same for `get`)

Reply:
Done in https://github.com/smithy-lang/smithy-rs/pull/3912/commits/0559e8b54fb41bcf04125e2e3cedeec87b24a9f9